linalg/constraints.cpp

```diff
@@ -92,50 +93,50 @@ EliminationProjection::EliminationProjection(const Operator& A,
 {
 }
 
-void EliminationProjection::Mult(const Vector& in, Vector& out) const
+void EliminationProjection::Mult(const Vector& vin, Vector& vout) const
 {
-   MFEM_ASSERT(in.Size() == width, "Wrong vector size!");
-   MFEM_ASSERT(out.Size() == height, "Wrong vector size!");
+   MFEM_ASSERT(vin.Size() == width, "Wrong vector size!");
+   MFEM_ASSERT(vout.Size() == height, "Wrong vector size!");
 
-   out = in;
+   vout = vin;
 
    for (int k = 0; k < eliminators.Size(); ++k)
    {
       Eliminator* elim = eliminators[k];
       Vector subvec_in;
       Vector subvec_out(elim->SecondaryDofs().Size());
-      in.GetSubVector(elim->PrimaryDofs(), subvec_in);
+      vin.GetSubVector(elim->PrimaryDofs(), subvec_in);
       elim->Eliminate(subvec_in, subvec_out);
-      out.SetSubVector(elim->SecondaryDofs(), subvec_out);
+      vout.SetSubVector(elim->SecondaryDofs(), subvec_out);
    }
 }
 
-void EliminationProjection::MultTranspose(const Vector& in, Vector& out) const
+void EliminationProjection::MultTranspose(const Vector& vin, Vector& vout) const
 {
-   MFEM_ASSERT(in.Size() == height, "Wrong vector size!");
-   MFEM_ASSERT(out.Size() == width, "Wrong vector size!");
+   MFEM_ASSERT(vin.Size() == height, "Wrong vector size!");
+   MFEM_ASSERT(vout.Size() == width, "Wrong vector size!");
 
-   out = in;
+   vout = vin;
 
    for (int k = 0; k < eliminators.Size(); ++k)
    {
       Eliminator* elim = eliminators[k];
       Vector subvec_in;
       Vector subvec_out(elim->PrimaryDofs().Size());
-      in.GetSubVector(elim->SecondaryDofs(), subvec_in);
+      vin.GetSubVector(elim->SecondaryDofs(), subvec_in);
       elim->EliminateTranspose(subvec_in, subvec_out);
-      out.AddElementVector(elim->PrimaryDofs(), subvec_out);
-      out.SetSubVector(elim->SecondaryDofs(), 0.0);
+      vout.AddElementVector(elim->PrimaryDofs(), subvec_out);
+      vout.SetSubVector(elim->SecondaryDofs(), 0.0);
    }
 }
 
 SparseMatrix * EliminationProjection::AssembleExact() const
 {
-   SparseMatrix * out = new SparseMatrix(height, width);
+   SparseMatrix * mout = new SparseMatrix(height, width);
```

Comment:
Is there a better name for this variable? What is the projection matrix typically called in the literature?

Reply:
Perhaps we can free up the name `mat` by renaming the eliminator specific matrix as `mat_k` where `k` is the loop iterator over eliminators.